CurrencyConverter/CurrencyConverter.Services/Services/CurrencyService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+using CurrencyConverter.Domain;
+
+namespace CurrencyConverter.Services.Services
+{
+    public class CurrencyService : ICurrencyService
+    {
+        public Currency GetCurrencyByCode(string code)
+        {
+            var list = GetCurrencyList();
+            var currency = list.FirstOrDefault(x => x.Code == code);
+            return currency;
+        }
+
+        private IEnumerable<Currency> GetECBList()
+        {
+            IList<Currency> list = new List<Currency> { new Currency { Code = "EUR", Rate = 1 } };
+            var doc = new XmlDocument();
+            doc.Load(@"http://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml");
+            var nodes = doc.SelectNodes("//*[@currency]");
+
+            if (nodes is null)
+            {
+                return list;
+            }
+
+            foreach (XmlNode node in nodes)
+            {
+                if (node.Attributes is null) continue;
+                var rate = new Currency
+                {
+                    Code = node.Attributes["currency"].Value,
+                    Rate = decimal.Parse(node.Attributes["rate"].Value, NumberStyles.Any, new CultureInfo("en-Us"))
+                };
+                list.Add(rate);
+            }
+
+            return list.ToArray();
+        }
+
+        public IEnumerable<Currency> GetCurrencyList()
```

Comment:
This basically calls methods that return rates from ECB and then symbols, and "joins" them into single list?

Reply:
Yup, that's exactly what it does